flow/layers/image_filter_layer.cc

```diff
@@ -11,6 +11,35 @@ ImageFilterLayer::ImageFilterLayer(sk_sp<SkImageFilter> filter)
       transformed_filter_(nullptr),
       render_count_(1) {}
 
+#ifdef FLUTTER_ENABLE_DIFF_CONTEXT
+
+void ImageFilterLayer::Diff(DiffContext* context, const Layer* old_layer) {
+  DiffContext::AutoSubtreeRestore subtree(context);
+  auto* prev = static_cast<const ImageFilterLayer*>(old_layer);
+  if (!context->IsSubtreeDirty()) {
+    FML_DCHECK(prev);
+    if (filter_ != prev->filter_) {
+      context->MarkSubtreeDirty(context->GetOldLayerPaintRegion(old_layer));
+    }
+  }
+
+  DiffChildren(context, prev);
+
+  SkMatrix inverse;
+  if (context->GetTransform().invert(&inverse)) {
+    auto paint_bounds = context->CurrentSubtreeRegion().GetBounds();
+    inverse.mapRect(&paint_bounds);
+    paint_bounds = filter_->computeFastBounds(paint_bounds);
+
+    context->AddPaintRegion(paint_bounds);
+    context->AddReadbackRegion(paint_bounds);
```

Comment:
ImageFilterLayer does no readback. It is downward filtering as opposed to backdropfilter which is upwards filtering.

Reply:
Technically it doesn't. But if you have blur children and you update child A, you need to repaint entire layer because you can't really clip blur. I mean in this case technically you can, but it would complicate things for first iteration so I decided to just redraw entire layer in that case.

Using readback here basically ensures that if any part of the layer needs to be repainted, whole layer will be repainted. This would not be necessary for example when using raster cache, but that's not always the case.

So right now the implementation is sub optional but should always have correct results.

Reply:
I would add a comment about that in the code.